flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/StreamSinkITCase.java

```diff
@@ -0,0 +1,221 @@
+				.addSink(TestSink
+						.newBuilder()
+						.setDefaultCommitter((Supplier<Queue<String>> & Serializable) () -> COMMIT_QUEUE)
+						.setGlobalCommitter((Supplier<Queue<String>> & Serializable) () -> GLOBAL_COMMIT_QUEUE)
+						.build());
+
+		env.execute();
+
+		assertThat(
+				COMMIT_QUEUE,
+				containsInAnyOrder(EXPECTED_COMMITTED_DATA_IN_BATCH_MODE.toArray()));
+
+		assertThat(
+				GLOBAL_COMMIT_QUEUE,
+				containsInAnyOrder(EXPECTED_GLOBAL_COMMITTED_DATA_IN_BATCH_MODE.toArray()));
+	}
+
+	@Test
+	public void writerAndCommitterExecuteInStreamingMode() throws Exception {
+		final StreamExecutionEnvironment env = buildStreamEnv();
+		final FiniteTestSource<Integer> source = new FiniteTestSource<>(SOURCE_DATA);
+
+		env.addSource(source, IntegerTypeInfo.INT_TYPE_INFO)
+				.addSink(TestSink
+						.newBuilder()
+						.setDefaultCommitter((Supplier<Queue<String>> & Serializable) () -> COMMIT_QUEUE)
+						.build());
+		env.execute();
+		assertThat(
+				COMMIT_QUEUE,
+				containsInAnyOrder(EXPECTED_COMMITTED_DATA_IN_STREAMING_MODE.toArray()));
+	}
+
+	@Test
+	public void writerAndCommitterExecuteInBatchMode() throws Exception {
+		final StreamExecutionEnvironment env = buildBatchEnv();
+
+		env.fromCollection(SOURCE_DATA)
+				.addSink(TestSink
+						.newBuilder()
+						.setDefaultCommitter((Supplier<Queue<String>> & Serializable) () -> COMMIT_QUEUE)
+						.build());
+		env.execute();
+		assertThat(
+				COMMIT_QUEUE,
+				containsInAnyOrder(EXPECTED_COMMITTED_DATA_IN_BATCH_MODE.toArray()));
+	}
+
+	@Test
+	public void writerAndGlobalCommitterExecuteInStreamingMode() throws Exception {
+		final StreamExecutionEnvironment env = buildStreamEnv();
+		final FiniteTestSource<Integer> source = new FiniteTestSource<>(SOURCE_DATA);
+
+		env.addSource(source, IntegerTypeInfo.INT_TYPE_INFO)
+				.addSink(TestSink
+						.newBuilder()
+						.setCommittableSerializer(TestSink.StringCommittableSerializer.INSTANCE)
+						.setGlobalCommitter((Supplier<Queue<String>> & Serializable) () -> GLOBAL_COMMIT_QUEUE)
+						.build());
+
+		env.execute();
+		assertThat(
+				GLOBAL_COMMIT_QUEUE,
+				containsInAnyOrder(EXPECTED_GLOBAL_COMMITTED_DATA_IN_STREAMING_MODE.toArray()));
+	}
+
+	@Test
+	public void writerAndGlobalCommitterExecuteInBatchMode() throws Exception {
+		final StreamExecutionEnvironment env = buildBatchEnv();
+
+		env.fromCollection(SOURCE_DATA)
+				.addSink(TestSink
+						.newBuilder()
+						.setCommittableSerializer(TestSink.StringCommittableSerializer.INSTANCE)
+						.setGlobalCommitter((Supplier<Queue<String>> & Serializable) () -> GLOBAL_COMMIT_QUEUE)
+						.build());
+		env.execute();
+
+		assertThat(
+				GLOBAL_COMMIT_QUEUE,
+				containsInAnyOrder(EXPECTED_GLOBAL_COMMITTED_DATA_IN_BATCH_MODE.toArray()));
+
+	}
+
+	private StreamExecutionEnvironment buildStreamEnv() {
+		final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		env.enableCheckpointing(100);
```

Comment:
We should also manually set the `RUNTIME_MODE` here be future proof in case the default changes.